lib/core/error.js

```diff
@@ -174,6 +175,20 @@ function isSDAMUnrecoverableError(error) {
   );
 }
 
+const SDAM_NODE_IS_SHUTTING_DOWN_ERRORS = new Set([
```

Comment:
Why this subset of errors and not all of the `recovering_codes` (`[11600, 11602, 13436, 189, 91]`) from the [spec](https://github.com/mongodb/specifications/blob/1e043d971e9f7793e5e5d4e68985a17313df643a/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#not-master-and-node-is-recovering)?